src/app/services/covid-data-api.service.ts

```diff
@@ -1,58 +1,97 @@
 import { Injectable, Inject, InjectionToken } from '@angular/core';
 import { CovidDataPoint } from '../models/CovidDataPoint';
+import { Country } from '../models/Country';
 
 import { Observable, EMPTY } from 'rxjs';
 import { catchError } from 'rxjs/operators';
 
-export interface Country {
-  // TODO: Move this somewhere where it makes more sense (maybe the country picker component)
-  id: number;
-  name: string;
-  iso2: string;
-  iso3: string;
-  continent: string;
-  lat: string;
-  long: string;
-}
-
-export interface CovidDataApiSubService {
+interface CovidDataApiServiceInterface {
   getLatestGlobalData(): Observable<CovidDataPoint>;
   getLatestCountryData(country: Country): Observable<CovidDataPoint>;
 }
 
+export interface CovidDataApiSubService extends CovidDataApiServiceInterface {}
+
 export const COVID_DATA_API_SUB_SERVICE = new InjectionToken<
   CovidDataApiSubService
 >('CovidDataApiSubService');
 
 @Injectable({
   providedIn: 'root',
 })
-export class CovidDataApiService {
+export class CovidDataApiService implements CovidDataApiServiceInterface {
   constructor(
     // Inspired by (https://stackoverflow.com/a/35916788)
     // Each api service must have its own provider registered in app.module for the InjectionToken
     @Inject(COVID_DATA_API_SUB_SERVICE)
-    private apiService: CovidDataApiSubService[]
+    private apiServices: CovidDataApiSubService[]
   ) {}
 
+  /**
+   * Cycles through each api service, calling the method provided, till one returns a value
+   * If all lead to errors, returns an empty observable
+   * @param {Observable<T>} method The method to be called
+   */
+  private getResultFromOneApiFor<T>(
+    method: (...args: any[]) => Observable<T>, // reference to a method with the same name being called
+    ...args: any[]
+  ): Observable<T> {
+    // if the method is not a class method of this class throw an error
+    if (
+      !Object.getOwnPropertyNames(CovidDataApiService.prototype).includes(
+        method.name
+      )
+    ) {
+      throw new Error(
+        `Method: ${method.name}() not found in Object: ${this.constructor.name}`
+      );
+    }
+    return this.apiServices.reduce<Observable<T>>((result, service) => {
+      return result !== EMPTY
+        ? result // result has already been found, skip
+        : service[method.name](...args) // call the method on the subservice, with the args
+            .pipe(
+              catchError((err) => {
+                this.handleError(err);
+                return EMPTY;
+              })
+            );
+    }, EMPTY);
```

Comment:
I've updated this to using `race` it sends out all requests and then just uses the first one that returns a good value. Think this would would be better overall instead of having to wait for them one by one. Although we might have more issues with some apis if they have any quota limits.